core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts

```diff
@@ -275,4 +297,45 @@ export class DotPaletteStore extends ComponentStore<DotPaletteState> {
         this.loadContentlets(variableName);
         this.setContentTypes(this.initialContent);
     }
+
+    /**
+     * Retrieves the experiment variant query field.
+     *
+     * @private
+     *
+     * @returns {string} The query field for the experiment variant.
+     */
+    private getExperimentVariantQueryField(): string {
+        return this.dotSessionStorageService.getVariationId() !== DEFAULT_VARIANT_ID
+            ? `+(variant:default OR variant:${this.dotSessionStorageService.getVariationId()})`
+            : '';
+    }
+
+    /**
+     * If the contentlets have a derivated/duplicated contentlet, remove the original (variant: DEFAULT).
+     *
+     * @param {DotCMSContentlet[]} contentlets - The array of contentlets to remove derived contentlets from.
+     * @return {DotCMSContentlet[]} - The modified array of contentlets without the original contentlets.
+     */
+    private removeDerivatedCotentlets(contentlets: DotCMSContentlet[]) {
+        const currentVariationId = this.dotSessionStorageService.getVariationId();
+
+        const uniqueIdentifiersFromVariantContentlet = new Set(
+            contentlets
+                .filter((item) => item.variant === currentVariationId)
+                .map((item) => item.identifier)
```

Comment:
This can be a `reduce` and with it we just need to go through the array once

Reply:
Maybe something like this:

```suggestion
                .reduce((acc, item) => {
                          if (item.variant === currentVariationId) {
                                 acc.push(item.identifier);
                          }
                          return acc;
                 }, []);
```